### DISCUSSION TITLE
How to group all updates that match package.json version range selectors - same behavior as `npm update`

### DISCUSSION THREAD

**Discussion by dominikg** [renovatebot/renovate#15174] [Request Help]
Example: https://github.com/sveltejs/vite-plugin-svelte/pull/313  where cookie is updated to 0.5.0 with `^0.4.0` in package.json

- Group [allnonmajor](https://docs.renovatebot.com/presets-group/#groupallnonmajor) includes it ( see https://github.com/renovatebot/renovate/issues/1257 why)

- Excluding packages where version starts with 0. altogether woud also prevent patch updates

- [Lockfilemaintenance](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance) reads like it could work, but there is no mention of pnpm and this won't bump package.json or allow any other kinds of filtering, right?

**Comment by rarkins** [comment]
Use matchCurrentVersion ">=1.0.0" for your package rule which groups non major updates. Then anything with a current 0.x will be excluded

**Reply by dominikg** [reply]
yes, but that would exclude 0.4.0 -> 0.4.1 aswell. I'm looking for the exact behavior of `npm update` with the caret selector.

**Reply by viceice** [reply]
use a second rule, match update type minor and current version less than 1.0.0 and set group to null, then minor updates are excluded from group.

**Reply by dominikg** [reply]
This should work for `^` and 0.x versions :+1: 

Is there a way to also get matching behavior for all of npm package range selectors so that a group includes every update that still satisfies package.json ranges? And could this be made available as a preset eg `group:allWithinRange`

**Reply by rarkins** [reply]
FYI all 0.x updates may be breaking

**Reply by dominikg** [reply]
I'm aware of that, but asking all packages in the npm registry to publish 1.x releases isn't feasible and some packages may even ignore semver altogether (again, see discussion in #1257) so rather than working with a strict semver interpretation that results in extra work i would prefer to work with npm's own tooling and restrict single packages by tighter version ranges in package.json if they prove to break `^` behavior.
